sway-lib-core/src/ops.sw

```diff
@@ -167,6 +167,7 @@ impl Not for bool {
 pub trait Eq {
     fn eq(self, other: Self) -> bool;
 } {
+    #[allow(dead_code)]
```

Comment:
While introducing this annotation in the compiler s quite useful and I'm on board with this PR, I'm a bit unsure about this particular line. The method `neq` here is a **public method in a library** and is no different than any other public free library function. Can we not simply **not** emit a warning for these types of methods?

Reply:
That's a good point. I will remove `neq` from the scope of this PR and fix #3674 in another PR that checks method visibility.